pkg/operator/encryption/state_test.go

```diff
@@ -0,0 +1,339 @@
+				keysRes := encryptionKeysResourceTuple{
+					resource: "secrets",
+					keys: []apiserverconfigv1.Key{
+						{
+							Name:   "34",
+							Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc=",
+						},
+
+						// secretsToProviders puts "fakeIdentityProvider" as last
+						{
+							Name:   "35",
+							Secret: newFakeIdentityEncodedKeyForTest(),
+						},
+					},
+				}
+				ec := createEncryptionCfgNoWriteKeyMultipleReadKeys([]encryptionKeysResourceTuple{keysRes})
+				return ec
+			}(),
+			output: map[schema.GroupResource]groupResourceKeys{
+				{Group: "", Resource: "secrets"}: {
+					writeKey: keyAndMode{
+						key: apiserverconfigv1.Key{Name: "", Secret: ""}, mode: "identity",
+					},
+					readKeys: []keyAndMode{
+						{key: apiserverconfigv1.Key{Name: "34", Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc="}, mode: "aescbc"},
+						{key: apiserverconfigv1.Key{Name: "35", Secret: newFakeIdentityEncodedKeyForTest()}, mode: "aescbc"},
+					},
+				},
+			},
+		},
+
+		// scenario 8
+		{
+			name: "turn off encryption for multiple resources",
+			input: func() *apiserverconfigv1.EncryptionConfiguration {
+				keysRes := []encryptionKeysResourceTuple{
+					{
+						resource: "secrets",
+						keys: []apiserverconfigv1.Key{
+							{
+								Name:   "34",
+								Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc=",
+							},
+
+							// secretsToProviders puts "fakeIdentityProvider" as last
+							{
+								Name:   "35",
+								Secret: newFakeIdentityEncodedKeyForTest(),
+							},
+						},
+					},
+
+					{
+						resource: "configmaps",
+						keys: []apiserverconfigv1.Key{
+							{
+								Name:   "34",
+								Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc=",
+							},
+
+							// secretsToProviders puts "fakeIdentityProvider" as last
+							{
+								Name:   "35",
+								Secret: newFakeIdentityEncodedKeyForTest(),
+							},
+						},
+					},
+				}
+				ec := createEncryptionCfgNoWriteKeyMultipleReadKeys(keysRes)
+				return ec
+			}(),
+			output: map[schema.GroupResource]groupResourceKeys{
+				{Group: "", Resource: "secrets"}: {
+					writeKey: keyAndMode{
+						key: apiserverconfigv1.Key{Name: "", Secret: ""}, mode: "identity",
+					},
+					readKeys: []keyAndMode{
+						{key: apiserverconfigv1.Key{Name: "34", Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc="}, mode: "aescbc"},
+						{key: apiserverconfigv1.Key{Name: "35", Secret: newFakeIdentityEncodedKeyForTest()}, mode: "aescbc"},
+					},
+				},
+
+				{Group: "", Resource: "configmaps"}: {
+					writeKey: keyAndMode{
+						key: apiserverconfigv1.Key{Name: "", Secret: ""}, mode: "identity",
+					},
+					readKeys: []keyAndMode{
+						{key: apiserverconfigv1.Key{Name: "34", Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc="}, mode: "aescbc"},
+						{key: apiserverconfigv1.Key{Name: "35", Secret: newFakeIdentityEncodedKeyForTest()}, mode: "aescbc"},
```

Comment:
this should be `aesgcm` - I need to fix that.

Reply:
Is this a `/hold`?

Reply:
ok, let's put it into `hold`, but please feel free to review it anyway.